### PR TITLE
Fix search highlight in selection anchor of cell selection

### LINKED ISSUE #14307: Search in selected cells does not work (highlight?) in the selection anchor cell
## Description

Search in selected cells works but the anchor of selection (the first cell from which the selection was extended) does not get searched/does not receive highlight:

## Reproduce

As on GIF:

Leading-cell-not-searched

## Expected behavior

The anchor cell is searched too.

## Context

- Browser and version: Chrome 111
- JupyterLab version: 4.0.0b0

### PR DESCRIPTION
**Pull request by krassowski**
## References

Fixes #14307.

## Code changes

- [x] resolves race condition between handlers of `activeCellChanged` and `selectionChanged` signals
- [x] adds a galata test

## User-facing changes

Search in selected cells works well when extending selection (covers all cells in selection).

## Backwards-incompatible changes

N/A

### CODE REVIEW

**Comment by krassowski** [comment]
bot please update galata snapshots

File: galata/test/jupyterlab/notebook-search.test.ts [hunk 1]
```diff
@@ -169,12 +169,51 @@ test.describe('Notebook Search', () => {
...
+
+    await page.waitForSelector('text=Search in 1 Selected Cell');
+
+    // Select cell above
+    await page.keyboard.press('Shift+ArrowUp');
+
+    // Expect updated text
+    await page.waitForSelector('text=Search in 2 Selected Cells');
+    // Expect 15 matches (note: should be 1/15, but this is xfail for now)
+    await page.waitForSelector('text=-/15');
```
Comment (krassowski) [inline galata/test/jupyterlab/notebook-search.test.ts:209]: This is addressed in https://github.com/jupyterlab/jupyterlab/pull/14387 but that part needs to wait for new release of CM6.

**Review by fcollonval — COMMENTED** [review]
The code looks good - thanks @krassowski 

I restarted the integration test to see if it has to do with flakiness or not.

**Review by fcollonval — APPROVED** [review]
Thanks @krassowski

**Pull request merged**